### PR TITLE
Store model outputs in specified directory

### PR DESCRIPTION
**Pull request by r-ash**
This PR will
* Allow a results dir to be passed to api call where model outputs will be saved

### CODE REVIEW

**Comment by r-ash** [comment]
Have verified this deploy works on dev system - you can see the version currently deployed runs the real model.

File: R/main.R [hunk 1]
```diff
@@ -3,18 +3,20 @@ main_api_args <- function(args = commandArgs(TRUE)) {
   hintr_api [options] [<queue_id>]
 
 Options:
---workers=N   Number of workers to spawn [default: 2]
---port=PORT   Port to use [default: 8888]"
+--workers=N              Number of workers to spawn [default: 2]
+--port=PORT              Port to use [default: 8888]
```
Comment (richfitz) [inline R/main.R]: please use `--results-dir=PATH` as the option (not the hyphen) and ensure to line up the spaces

File: R/main.R [hunk 2]
```diff
@@ -3,18 +3,20 @@ main_api_args <- function(args = commandArgs(TRUE)) {
...
+--results_dir=RESULT_DIR  Directory to store model results in [default: results/]"
 
   dat <- docopt::docopt(usage, args)
   list(port = as.integer(dat$port),
        queue_id = dat$queue_id,
-       workers = as.integer(dat$workers))
+       workers = as.integer(dat$workers),
+       results_dir = dat$results_dir)
```
Comment (richfitz) [inline R/main.R]: this will be accessed as `dat[["results-dir"]]` I believe

File: R/run_model.R [hunk 3]
```diff
@@ -34,7 +34,12 @@ run_model <- function(data, options) {
          spectrum_path = system_file("output", "malawi_spectrum_download.zip"),
          summary_path = system_file("output", "malawi_summary_download.zip")))
   }
-  naomi::hintr_run_model(data, options)
+  dir <- normalizePath(dir, mustWork = TRUE)
+  output_path <- tempfile(tmpdir = dir, fileext = ".RDS")
```
Comment (richfitz) [inline R/run_model.R]: why are we using shouty case here? `.rds` is more typical

File: tests/testthat/test-main.R [hunk 4]
```diff
@@ -2,10 +2,16 @@ context("main")
 
 test_that("main_api_args", {
   expect_equal(main_api_args(c()),
-               list(port = 8888, queue_id = NULL, workers = 2))
+               list(port = 8888, queue_id = NULL, workers = 2,
+                    results_dir = "results/"))
```
Comment (richfitz) [inline tests/testthat/test-main.R]: no trailing slash

File: R/run_model.R (hunk 3 above)
Comment (r-ash) [reply R/run_model.R]: No good reason - I have fixed that.

File: tests/testthat/test-main.R (hunk 4 above)
Comment (r-ash) [reply tests/testthat/test-main.R]: Fixed

File: R/main.R (hunk 1 above)
Comment (r-ash) [reply R/main.R]: Fixed that too

**Pull request merged**